src/DIRAC/FrameworkSystem/Service/BundleDeliveryHandler.py

```diff
@@ -148,24 +137,22 @@ def __transferFile(self, filetype, fileHelper):
         elif filetype == "CRLs":
             retVal = Utilities.generateRevokedCertsFile()
         else:
-            return S_ERROR("Not supported file type %s" % filetype)
+            return S_ERROR(f"Not supported file type {filetype}")
 
         if not retVal["OK"]:
             return retVal
-        else:
-            result = fileHelper.getFileDescriptor(retVal["Value"], "r")
-            if not result["OK"]:
-                result = fileHelper.sendEOF()
-                # better to check again the existence of the file
-                if not os.path.exists(retVal["Value"]):
-                    return S_ERROR("File %s does not exist" % os.path.basename(retVal["Value"]))
-                else:
-                    return S_ERROR("Failed to get file descriptor")
-            fileDescriptor = result["Value"]
-            result = fileHelper.FDToNetwork(fileDescriptor)
-            fileHelper.oFile.close()  # close the file and return
-            return result
+        result = fileHelper.getFileDescriptor(retVal["Value"], "r")
```

Comment:
I am super confused if this should be `"rb"` ...

Reply:
I am also confused. In doubt, I don't change what's there.

Reply:
It definitely "should" be `"rb"`. If it's currently working it's either:
* using surrogate escapes
* the opened files only contain valid unicode sequences

That said, I also lean towards not touching it if we don't have a reason to think it's broken.